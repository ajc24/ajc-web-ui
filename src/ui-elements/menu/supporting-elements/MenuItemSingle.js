import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../../../css/common.css';
import '../../../css/menu.css';

/**
 * The MenuItemSingle component acts as a standard menu item for the MenuBar component
 */
class MenuItemSingle extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const rootStyling = 'ajc-flex-row ajc-justify-left';
    let contentStyling = 'ajc-flex-row ajc-menu-item';
    if (this.props.colour === 'red') {
      contentStyling += ' ajc-menu-item-red';
    } else {
      contentStyling += ' ajc-menu-item-grey';
    }
    const linkStyling = 'ajc-flex-row ajc-menu-item-content ajc-font-menu ajc-text-white';
    return (
      <div id={`${this.props.id}-root`} className={rootStyling}>
        <div id={this.props.id} className={contentStyling}>
          <Link to={this.props.route} className={linkStyling} tabIndex={0}>
            &nbsp;&nbsp;{this.props.title}&nbsp;&nbsp;
          </Link>
        </div>
      </div>
    );
  }
}
MenuItemSingle.propTypes = {
  /** The components unique identifier. */
  id: PropTypes.string.isRequired,
  /** The background colour for the menu item. By default this is set to grey. */
  colour: PropTypes.oneOf([ 'grey', 'red' ]),
  /** The route to navigate to on clicking on the menu item. */
  route: PropTypes.string.isRequired,
  /** The title for the menu item being displayed. */
  title: PropTypes.string.isRequired,
};
MenuItemSingle.defaultProps = {
  colour: 'grey'
};
export default MenuItemSingle;
