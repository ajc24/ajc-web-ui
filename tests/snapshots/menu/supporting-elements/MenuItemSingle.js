import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { BrowserRouter } from 'react-router-dom';
import MenuItemSingle from '../../../../src/ui-elements/menu/supporting-elements/MenuItemSingle';

describe('MenuItemSingle', () => {
  describe('Default props and rendering', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="navigation">
          <BrowserRouter>
            <MenuItemSingle id="test-menu-item-single" route="/test-route" title="Menu Item (Single)"/>
          </BrowserRouter>
        </div>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering: MenuItemSingle in red', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="navigation">
          <BrowserRouter>
            <MenuItemSingle id="test-menu-item-single" route="/test-route" title="Menu Item (Single)" colour="red"/>
          </BrowserRouter>
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
