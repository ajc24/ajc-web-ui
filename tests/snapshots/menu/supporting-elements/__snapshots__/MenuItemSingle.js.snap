// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemSingle Default props and rendering verifies the snapshot for the component 1`] = `
<div
  role="navigation"
>
  <div
    className="ajc-flex-row ajc-justify-left"
    id="test-menu-item-single-root"
  >
    <div
      className="ajc-flex-row ajc-menu-item ajc-menu-item-grey"
      id="test-menu-item-single"
    >
      <a
        className="ajc-flex-row ajc-menu-item-content ajc-font-menu ajc-text-white"
        href="/test-route"
        onClick={[Function]}
        tabIndex={0}
      >
          
        Menu Item (Single)
          
      </a>
    </div>
  </div>
</div>
`;

exports[`MenuItemSingle Transferred props and rendering: MenuItemSingle in red verifies the snapshot for the component 1`] = `
<div
  role="navigation"
>
  <div
    className="ajc-flex-row ajc-justify-left"
    id="test-menu-item-single-root"
  >
    <div
      className="ajc-flex-row ajc-menu-item ajc-menu-item-red"
      id="test-menu-item-single"
    >
      <a
        className="ajc-flex-row ajc-menu-item-content ajc-font-menu ajc-text-white"
        href="/test-route"
        onClick={[Function]}
        tabIndex={0}
      >
          
        Menu Item (Single)
          
      </a>
    </div>
  </div>
</div>
`;
